lib/subjects/postgresql_selection.rb

```diff
@@ -29,44 +29,8 @@ def available
                           # default mode: do not select from training sets
                           workflow.non_training_subject_sets.pluck(:id)
                         end
-      if Gem::Version.new(Rails.version) > Gem::Version.new('5.0') && Gem::Version.new(Rails.version) < Gem::Version.new('5.1')
-        # Handle a bug when the rails 5.0 AR scope bind params merge in rails 5.0 (works fine in 4.2)
-        #
-        # failed fix 1 - ensure the subject_set_id param clause is before the id clause
-        #
-        # SetMemberSubject.where(subject_set_id: subject_set_ids, id: query).select(:id)
-        #
-        # as that fixes the incorrect subject_set_id bind param - fails to bind limit correctly later on :(
-        #
-        # failed fix 2 - leave as is but wrap subject set ids in array
-        #
-        # query.where(subject_set_id: subject_set_ids)
-        #
-        # duplicate query clauses which make no sense and can be cleaned up
-        # but it still fails on the limit bind param fails later on :(
-        #
-        # failed fix 3 - tried to merge the AR queries manually
-        #
-        # query.merge(SetMemberSubject.where(subject_set_id: subject_set_ids))
-        #
-        # fixes the duplicate query clause but still fails on the limit bind param later on :(
-        #
-        # failed fix 4 - use an explicit sub query
-        #
-        # SetMemberSubject.where(subject_set_id: subject_set_ids, id: SetMemberSubject.select('id').from(query)).select(:id)
-        #
-        # fixes the subject set id bind param but still fails on the limit bind param later on :(
 
-        # Working solution
-        #
-        # run two queries
-        # one with with a pluck on the query select to pull the avaialble SMS IDs
-        # and then apply the subject set id filter to create a new simple scope
-        # that we pass onto to the selector strategy class which then applies the limit bind params etc
         SetMemberSubject.where(id: query.pluck(:id), subject_set_id: subject_set_ids).select(:id)
```

Comment:
Unfortunately, i tried the other solutions mentioned and it still fails in 5.1. we could keep what we had but update the check to run `if Gem::Version.new(Rails.version) > Gem::Version.new('5.0') && Gem::Version.new(Rails.version) < Gem::Version.new('5.2')`.

Reply:
cool - that makes sense then - perhaps leave these comments in and the Gem Version comparison switch to remind us to try this later...

fwiw - i looked into this again and i think we're using too many layers of query merging. TBH this code could do with a refactor but that's a lot of work and using 2 trips here simplifies the resulting query and our understanding of it so that's probably a win for us maintaining this code.  

As you were :) ii